wkcuber/api/MagDataset.py

```diff
@@ -0,0 +1,135 @@
+from os.path import join
+
+from wkw import wkw
+
+import wkcuber.api as api
+from wkcuber.api.Slice import WKSlice, TiffSlice
+from wkcuber.api.TiffData.TiffMag import TiffMagHeader
+
+
+class MagDataset:
+    def __init__(self, layer, name):
+        self.mag = {}
+        self.layer = layer
+        self.name = name
+        self.header = self.get_header()
+
+        file_path = join(self.layer.dataset.path, self.layer.name, self.name)
+        size = self.layer.dataset.properties.data_layers[
+            self.layer.name
+        ].get_bounding_box_size()
+        offset = self.layer.dataset.properties.data_layers[
+            self.layer.name
+        ].get_bounding_box_offset()
+        self.slice = self.__get_slice_type__()(
+            file_path, self.header, size, global_offset=(0, 0, 0)
+        )
+
+        # MagDataset uses Slice to access the actual data, but there are no bounds for an MagDataset (unlike a Slice)
+        self.slice.is_bounded = False
+
+    def open(self):
+        self.slice.open()
+
+    def close(self):
+        self.slice.close()
+
+    def read(self, size, offset=(0, 0, 0)):
+        return self.slice.read(size, offset)
+
+    def write(self, data, offset=(0, 0, 0)):
+        self.__assert_valid_num_channels__(data.shape)
+        self.slice.write(data, offset)
+        layer_properties = self.layer.dataset.properties.data_layers[self.layer.name]
+        current_offset = layer_properties.get_bounding_box_offset()
+        current_size = layer_properties.get_bounding_box_size()
+
+        if self.slice.check_bounds(current_offset, current_size):
+            new_offset = (
+                offset
+                if current_offset == (-1, -1, -1)
+                else tuple(min(x) for x in zip(current_offset, offset))
+            )
+            total_size = tuple(max(x) for x in zip(current_size, data.shape[-3:]))
+            self.slice.size = total_size
+            # no need to update the offset of the slice, because it is always the absolute offset (0, 0, 0)
+
+            self.layer.dataset.properties.set_bounding_box_size_of_layer(
```

Comment:
Maybe provide a method like `set_bounding_box_of_layer` to which you pass `total_size` and `new_offset`. This makes the code here more concise and the properties are only updated once on disk, if I understand this correctly.